index.js

```diff
@@ -5,15 +5,47 @@
 
 // This script is modified version of
 // https://github.com/brave/brave-ios/blob/development/Client/Frontend/UserContent/UserScripts/Playlist.js
-(function () {
+medias = (async function () {
   // This will be replaced by native code on demand.
   const siteSpecificDetector = null
 
+  async function isMediaSourceObjectURL (src) {
+    if (!src || !src.startsWith('blob:')) {
+      return false
+    }
+
+    const controller = new AbortController()
+    const signal = controller.signal
+    let timeout
+    const maybeAbortFetch = new Promise(resolve =>
+      timeout = setTimeout(() => resolve(false), 500)
+    )
+
+    return Promise.any([
+      new Promise(resolve => {
+        fetch(src, {
+          signal
+        }).then(response => {
+          resolve(false)
+        }).catch(() => {
+          resolve(true)
```

Comment:
Can you share something more why it can return `false` and `true` from `then` and `catch` clause? want to study :)

Reply:
Ah, blob url is only valid in user agent. so it gets valid response if it's not blob?

Reply:
A `blob: url` could be pointing at `MediaSource` or `Blob` object reside in Blob storage. In case `MediaSource`, fetch will fail as the source. On the other hand, if the url is for `Blob` object, fetch will succeed in getting data.

```javascript
let blob = new Blob([new ArrayBuffer(data)], { type: "image/png" });
const url = window.URL.createObjectURL(blob); // blob:http://foo.bar
```

@fallaciousreasoning @szilardszaloki Please correct me if I misunderstand logic behind this 😄

Reply:
Yup, exactly - trying to fetch a `Blob` url which points to a MediaSource will cause an exception. Honestly, it's a bit annoying that MediaSource can be converted to blob urls, as they're handled completely differently (as above, you can't even fetch them).

Reply:
In addition, a `blob:` backed by a `File` is fetchable, too. Basically, the approach is to see if the `blob:` is backed by a `MediaSource`, or not. Bear in mind, though, that this is only reliable if `fetch()` doesn't reuse this error condition (`TypeError` with the message `Failed to fetch`) in other `blob:` scenarios. If it does, we're out of luck (and should probably move to `MAIN`, and monkey patch `URL.createObjectURL()`). Nevertheless, this works in `ISOLATED`, and should be a good approximation.